### ISSUE TITLE
vpc generation=2 ibm_is_security_group_rule remote to another security group fails

### ISSUE THREAD

**Issue by powellquiring** [IBM-Cloud/terraform-provider-ibm#746]
On generation=1 the terraform config creates the following resources:
```
~/tmp/bug $ tfa
data.ibm_resource_group.all_rg: Refreshing state...
ibm_is_vpc.vpc: Creating...
  classic_access:         "" => "false"
  default_network_acl:    "" => "<computed>"
  default_security_group: "" => "<computed>"
  is_default:             "" => "false"
  name:                   "" => "tfbvpc-pubpriv"
  resource_group:         "" => "5e9df23ddbe04b6fa9ba3d3217ccd51e"
  status:                 "" => "<computed>"
ibm_is_vpc.vpc: Still creating... (10s elapsed)
ibm_is_vpc.vpc: Creation complete after 20s (ID: b6d466fd-b9a6-41f9-9854-f12d048e0ca8)
ibm_is_security_group.bastion: Creating...
  name:           "" => "tfbvpc-pubpriv-bastion-sg"
  resource_group: "" => "<computed>"
  rules.#:        "" => "<computed>"
  vpc:            "" => "b6d466fd-b9a6-41f9-9854-f12d048e0ca8"
ibm_is_security_group.maintenance: Creating...
  name:           "" => "tfbvpc-pubpriv-maintenance-sg"
  resource_group: "" => "5e9df23ddbe04b6fa9ba3d3217ccd51e"
  rules.#:        "" => "<computed>"
  vpc:            "" => "b6d466fd-b9a6-41f9-9854-f12d048e0ca8"
ibm_is_security_group.maintenance: Creation complete after 3s (ID: 2d364f0a-a870-42c3-a554-000001994216)
ibm_is_security_group.bastion: Creation complete after 4s (ID: 2d364f0a-a870-42c3-a554-000001994060)
ibm_is_security_group_rule.maintenance_ingress_ssh_bastion: Creating...
  direction:      "" => "inbound"
  group:          "" => "2d364f0a-a870-42c3-a554-000001994216"
  ip_version:     "" => "ipv4"
  remote:         "" => "2d364f0a-a870-42c3-a554-000001994060"
  rule_id:        "" => "<computed>"
  tcp.#:          "" => "1"
  tcp.0.port_max: "" => "22"
  tcp.0.port_min: "" => "22"
ibm_is_security_group_rule.maintenance_ingress_ssh_bastion: Creation complete after 2s (ID: 2d364f0a-a870-42c3-a554-000001994216.b597cff2-38e8-4e6e-999d-000006188396)

Apply complete! Resources: 4 added, 0 changed, 0 destroyed.
```
On generation=2:
```
~/tmp/bug $ tfa
data.ibm_resource_group.all_rg: Refreshing state...
ibm_is_vpc.vpc: Creating...
  classic_access:         "" => "false"
  default_network_acl:    "" => "<computed>"
  default_security_group: "" => "<computed>"
  is_default:             "" => "false"
  name:                   "" => "tfbvpc-pubpriv"
  resource_group:         "" => "5e9df23ddbe04b6fa9ba3d3217ccd51e"
  status:                 "" => "<computed>"
ibm_is_vpc.vpc: Still creating... (10s elapsed)
ibm_is_vpc.vpc: Creation complete after 12s (ID: r006-faf90de4-9c5f-4799-acac-df95b470db7c)
ibm_is_security_group.bastion: Creating...
  name:           "" => "tfbvpc-pubpriv-bastion-sg"
  resource_group: "" => "<computed>"
  rules.#:        "" => "<computed>"
  vpc:            "" => "r006-faf90de4-9c5f-4799-acac-df95b470db7c"
ibm_is_security_group.maintenance: Creating...
  name:           "" => "tfbvpc-pubpriv-maintenance-sg"
  resource_group: "" => "5e9df23ddbe04b6fa9ba3d3217ccd51e"
  rules.#:        "" => "<computed>"
  vpc:            "" => "r006-faf90de4-9c5f-4799-acac-df95b470db7c"
ibm_is_security_group.maintenance: Creation complete after 0s (ID: r006-13d9a314-a28d-43ae-90ef-f5d8ba0fdf81)
ibm_is_security_group.bastion: Creation complete after 1s (ID: r006-fd32a60e-554e-4ca3-b4d1-6d39a7b8bee2)

Error: Error applying plan:

1 error occurred:
	* ibm_is_security_group_rule.maintenance_ingress_ssh_bastion: "remote" (r006-fd32a60e-554e-4ca3-b4d1-6d39a7b8bee2) invalid security group remote

```

configuration:
```
provider "ibm" {
  region           = "${var.region}"
  ibmcloud_api_key = "${var.ibmcloud_api_key}"
  generation       = "${var.generation}"
}


locals {
  BASENAME = "${var.prefix}vpc-pubpriv"
}

data "ibm_resource_group" "all_rg" {
  name = "${var.resource_group_name}"
}

resource "ibm_is_vpc" "vpc" {
  name           = "${local.BASENAME}"
  resource_group = "${data.ibm_resource_group.all_rg.id}"
}

resource "ibm_is_security_group" "bastion" {
  name = "${local.BASENAME}-bastion-sg"
  vpc            = "${ibm_is_vpc.vpc.id}"
}

resource "ibm_is_security_group" "maintenance" {
  name           = "${local.BASENAME}-maintenance-sg"
  vpc            = "${ibm_is_vpc.vpc.id}"
  resource_group = "${data.ibm_resource_group.all_rg.id}"
}

resource "ibm_is_security_group_rule" "maintenance_ingress_ssh_bastion" {
  group     = "${ibm_is_security_group.maintenance.id}"
  direction = "inbound"
  remote    = "${ibm_is_security_group.bastion.id}"

  tcp = {
    port_min = 22
    port_max = 22
  }
}
```
### Terraform Version
~/tmp/bug $ tf version
Terraform v0.11.14
+ provider.ibm v0.17.3

Your version of Terraform is out of date! The latest version
is 0.12.9. You can update by downloading from www.terraform.io/downloads.html

**Comment by i-am-david** [comment]
@hkantare Could we prioritize this please for VPC NG GA? per Jennifer their GA is mid-October and this is holding up their solution tutorials. 

![image](https://user-images.githubusercontent.com/50240354/66053389-d5691700-e4f7-11e9-9877-2199993ad98e.png)

FYI @davetropeano

**Comment by hkantare** [comment]
Yes we are working on this..

**Comment by hkantare** [comment]
Fixed in v0.17.5 release
https://github.com/IBM-Cloud/terraform-provider-ibm/releases/tag/v0.17.5

**Comment by powellquiring** [comment]
verified
